src/benchmark_runner.cc

```diff
@@ -140,23 +145,110 @@ void RunInThread(const BenchmarkInstance* b, IterationCount iters,
   manager->NotifyThreadComplete();
 }
 
+BenchTimeType ParseBenchMinTime(const std::string& value) {
+  BenchTimeType ret;
+
+  if (value.empty()) {
+    ret.tag = BenchTimeType::TIME;
+    ret.time = 0.0;
+    return ret;
+  }
+
+  if (value.back() == 'x') {
+    const char* iters_str = value.c_str();
+    char* p_end;
+    // Reset error since atol will change it.
+    const int old_errno = errno;
+    errno = 0;
+    IterationCount num_iters = std::strtol(iters_str, &p_end, 10);
+    const int new_errno = errno;
+    errno = old_errno;
+
+    (void)new_errno;
+
+    // After a valid parse, p_end should have been set to
+    // point to the 'x' suffix.
+    BM_CHECK(new_errno == 0 && p_end != nullptr && *p_end == 'x')
+        << "Malformed iters value passed to --benchmark_min_time: `" << value
+        << "`. Expected --benchmark_min_time=<integer>x.";
+
+    ret.tag = BenchTimeType::ITERS;
+    ret.iters = num_iters;
+    return ret;
+  }
+
+  const char* time_str = value.c_str();
+  bool has_suffix = value.back() == 's';
+  if (!has_suffix) {
+    BM_VLOG(0) << "Value passed to --benchmark_min_time should have a suffix. "
+                  "Eg., `30s` for 30-seconds.";
+  }
+
+  char* p_end;
+  // Reset error before it's changed by strtod.
+  const int old_errno = errno;
+  errno = 0;
+  double min_time = std::strtod(time_str, &p_end);
+  const int new_errno = errno;
+  errno = old_errno;
```

Comment:
Why do we care to restore the `errno`?

Reply:
just thought it's generally good to not pollute errno - can remove this if you'd prefer

Reply:
Why not std::stod()?

Reply:
because it didn't have a very good error-reporting interface (see previous iteration of this patch where stod was used)